HikVisionSource/src/main/java/io/vantiq/extsrc/HikVisionSource/HikVisionCore.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (c) 2020 Vantiq, Inc.
+ *
+ * All rights reserved.
+ * 
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.HikVisionSource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Timer;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vantiq.extjsdk.ExtensionServiceMessage;
+import io.vantiq.extjsdk.ExtensionWebSocketClient;
+import io.vantiq.extjsdk.Handler;
+import io.vantiq.extsrc.HikVisionSource.exception.*;
+
+/**
+ * Controls the connection and interaction with the Vantiq server. Initialize it
+ * and call start() and it will run itself. start() will return a boolean
+ * describing whether or not it succeeded, and will wait up to 10 seconds if
+ * necessary.
+ */
+public class HikVisionCore {
+
+    private static final String SYNCH_LOCK = "synchLock";
+
+    String sourceName;
+    String authToken;
+    String targetVantiqServer;
+
+    HikVisionHandleConfiguration oConfigHandler;
+
+    Timer pollTimer = null;
+    ExtensionWebSocketClient client = null;
+    HikVision hikVision = null;
+
+    ExecutorService publishPool = null;
+
+    final Logger log;
+    final static int RECONNECT_INTERVAL = 5000;
+
+    /**
+     * Executes the query that is provided in the Publish Message. If query is an
+     * Array of Strings, then it is executed as a Batch request. If the query is a
+     * single String, then it is executed normally.
+     * 
+     * @param message The Query message.
+     */
+    public void executePublish(ExtensionServiceMessage message) {
+        Map<String, ?> request = (Map<String, ?>) message.getObject();
+
+        // Getting local copy of HikVision class
+        HikVision localHikVision = null;
+        synchronized (SYNCH_LOCK) {
+            localHikVision = hikVision;
+        }
+
+        if (localHikVision == null) {
+            log.error("EasyModbus connection closed before operation could complete");
+        }
+
+        // Gather query results, or send a query error if an exception is caught
+        try {
+            int data = localHikVision.hanldeUpdateCommand(message);
```

Comment:
`int data =` is unnecessary as the data variable isn't accessible out of the try block.

Reply:
done